aten/src/ATen/native/native_functions.yaml

```diff
@@ -6642,6 +6642,16 @@
   structured_delegate: _addmm_activation.out
   variants: function, method
 
+- func: _scaled_mm(Tensor self, Tensor mat2, *, ScalarType? dtype=None, Tensor? scale_a=None, Tensor? scale_b=None, Tensor? scale_result=None) -> (Tensor, Tensor)
+  variants: function
+  dispatch:
+    CUDA: _scaled_mm_cuda
+
+- func: _scaled_mm.out(Tensor self, Tensor mat2, *, ScalarType? dtype=None, Tensor? scale_a=None, Tensor? scale_b=None, Tensor? scale_result=None, Tensor(a!) out, Tensor(b!) out_amax) -> (Tensor(a!), Tensor(b!))
```

Comment:
I would suggest you do this as a structured kernel (similar to addmm today). You only have to replace this with a meta impl (which is trivial) the right way. This will allow you to remove a bunch of manual cpp code.

Reply:
Hmm, are you saying that structured kernels will take PyThon meta implementation?